### PR TITLE
Return error if tokens cannot be deleted because they do not exist

### PR DESCRIPTION
**Pull request by chelseakomlo**
Fixes issue #3238

### CODE REVIEW

File: nomad/acl_endpoint.go [hunk 1]
```diff
@@ -578,12 +579,14 @@ func (a *ACL) DeleteTokens(args *structs.ACLTokenDeleteRequest, reply *structs.G
...
 	hasGlobal := false
 	allGlobal := true
+	nonexistentTokens := make([]string, 0)
 	for _, accessor := range args.AccessorIDs {
 		token, err := state.ACLTokenByAccessorID(nil, accessor)
 		if err != nil {
 			return fmt.Errorf("token lookup failed: %v", err)
 		}
-		if token == nil {
+		if token == nil { // why continue if the token is nil???
```
Comment (schmichael) [inline nomad/acl_endpoint.go]: Seems like you can remove this comment as it appears the reason we continue is to return a complete list of missing tokens (which is nice from a UX perspective).

File: nomad/acl_endpoint.go [hunk 2]
```diff
@@ -593,6 +596,10 @@ func (a *ACL) DeleteTokens(args *structs.ACLTokenDeleteRequest, reply *structs.G
 		}
 	}
 
+	if len(nonexistentTokens) != 0 {
+		return fmt.Errorf("Cannot delete nonexistant tokens: %v", strings.Join(nonexistentTokens, ", "))
```
Comment (dadgar) [inline nomad/acl_endpoint.go]: nonexistant -> nonexistent

Comment (dadgar) [reply nomad/acl_endpoint.go]: Fix in test too

File: nomad/acl_endpoint.go (hunk 1 above)
Comment (chelseakomlo) [reply nomad/acl_endpoint.go]: Yes, thank you! That is leftover from development notes, it shouldn't be merged

**Pull request merged**